terraform-docker/apply/main.yaml

```diff
@@ -52,6 +61,55 @@ stages:
                         artifact: "${{ env.name }}.enc"
                         path: $(Build.Repository.LocalPath)/${{ parameters.terraformFolder }}/.terraform/plans
 
+                    - task: AmazonWebServices.aws-vsts-tools.AWSShellScript.AWSShellScript@1
+                      displayName: "Get AWS credentials"
+                      inputs:
+                        workingDirectory: "."
+                        awsCredentials: ${{ format(parameters.awsServiceConnectionTemplate, env.name) }}
+                        regionName: ${{ parameters.awsRegion }}
+                        scriptType: inline
+                        inlineScript: |
+                          set -e
+                          AWS_CONFIG_FILE=${HOME}/.aws/config
+                          AWS_SHARED_CREDENTIALS_FILE=${HOME}/.aws/credentials
+                          TMPFILE=${HOME}/.aws/tmp.json
+                          mkdir $(dirname $TMPFILE)
+                          aws sts get-federation-token --name xkf-$ENV --policy-arns arn="$AWS_ARN" --region $AWS_REGION > $TMPFILE
```

Comment:
Can't the ARN be extracted from the environment variables (should be available either directly because of the task or available through an AWS cli command)?

Reply:
I think it might be possible to use `aws sts get-caller-identity |jq -r .Arn` but at the same time do we always want to use the max amount of access that we have?
Of course It will make it easier for the user.
Could add some logic to have current arn as default but give the possibility to overwrite it.

Reply:
So I gave it a try and to my knowledge it's not possible.

```shell
aws sts get-caller-identity |jq -r .Arn
# returns something like
arn:aws:iam::ACCOUNTID:user/folder/name
```

You can do something like https://docs.aws.amazon.com/cli/latest/reference/iam/get-user-policy.html but you have to provide the name of the policy so it won't help you.
So we sadly have to provide the ARN.